service/gcsutils/tarlib/tardisk.go

```diff
@@ -126,7 +126,7 @@ func CreateTarDisk(in io.Reader,
 
 	logrus.Info("entering CreateTarDisk")
 
-	mntFolder, err := ioutil.TempDir(tmpdir, "mnt")
```

Comment:
I did try running gcstools bits without the changes in this PR and I didn't see the original write error that I was seeing. I might have messed up my environment. I will close this PR for now. And re-open it again with appropriate changes if it's hit again.